File: GC.Domain/Records/Novelties/Novelty.cs
using System;

namespace GC.Domain.Records.Novelties
{
    public class Novelty
    {
        public Guid Id { get; }
        public String Title { get; }
        public String Description { get; }
        public DateTime? PublishDate { get; }
        public String? Image { get; }

        public Novelty(Guid id, String title, String description, DateTime? publishDate, String? image)
        {
            Id = id;
            Title = title;
            Description = description;
            PublishDate = publishDate;
            Image = image;
        }
    }
}

File: GC.WebSpace/ClientApp/domain/records/novelties/novelty.ts

export class Novelty {
    constructor(
        public readonly id: string,
        public readonly title: string,
        public readonly description: string,
        public readonly publishDate: Date | null,
        public readonly image: string | null
    ) { }

    public get shortDescription() {
        return this.description.substring(0, 90) + "...";
    }
}

export namespace Novelty {
    export function toModel(object: any): Novelty {
        return new Novelty(object.id, object.title, object.description, object.publishDate == null ? null : new Date(object.publishDate), object.image);
    }

    export function toModels(objects: any[]): Novelty[] {
        return objects.map(toModel);
    }
}